compiler/src/openxla/compiler/nvgpu/Conversion/ConvertHLOToCUDNN.cpp

```diff
@@ -125,14 +130,29 @@ static LogicalResult outlineToGraph(TensorValue result,
       result.getLoc(), funcType.getResults(), results);
   rewriter.create<ReturnOp>(root->getLoc(), castOp.getResults());
 
+  // Create global handle before any other initialization.
+  rewriter.setInsertionPointToStart(m.getBody());
+  Location loc = root->getLoc();
+  std::string globalHandleName = (graph.getName() + ".handle").str();
+  auto handleTy = rewriter.getType<HandleType>();
+  auto globalHandle = rewriter.create<IREE::Util::GlobalOp>(
+      loc, globalHandleName, /*isMutable=*/false, handleTy);
+
+  // Create global handle initializer.
+  auto initializer = rewriter.create<IREE::Util::InitializerOp>(loc);
+  rewriter.setInsertionPointToStart(initializer.addEntryBlock());
+  auto device = rewriter.create<IREE::HAL::ExSharedDeviceOp>(loc);
+  auto handle = rewriter.create<HandleOp>(loc, device);
+  rewriter.create<IREE::Util::GlobalStoreOp>(loc, handle, globalHandle);
+  rewriter.create<IREE::Util::InitializerReturnOp>(loc);
+
   // Replace root with cudnn.call op.
   rewriter.setInsertionPoint(root);
-  auto device = rewriter.create<IREE::HAL::ExSharedDeviceOp>(root->getLoc());
-  auto handle = rewriter.create<HandleOp>(root->getLoc(), device.getResult());
+  auto loadedHandle = rewriter.create<IREE::Util::GlobalLoadOp>(
+      loc, rewriter.getType<HandleType>(), globalHandle.getSymNameAttr());
```

Comment:
`handleTy` instead of getType? or getType everywhere

Reply:
Done.